packages/stream-tile/src/tile-document.ts

```diff
@@ -370,13 +373,14 @@ export class TileDocument<T = Record<string, any>> extends Stream {
     }
 
     const patch = jsonpatch.compare(this.content, newContent)
-    const commit: RawCommit = {
+    const genesisLogEntry = this.state.log[0]
+    if (!genesisLogEntry) throw new Error(`No genesis log entry`)
```

Comment:
given this function isn't static, but is called on an actual TileDocument instance, and the fact that you can't construct an actual TileDocument instance if there's no genesis commit, then it seems like this check shouldn't be necessary?

Reply:
IMO, this is exactly pitfall in reasoning highlighted by TS now.

See for example this code in temp file `./packages/http-client/src/foo.ts`.
```
import { AnchorStatus, SignatureStatus, StreamState, TestUtils } from '@ceramicnetwork/common'
import { TileDocument } from '@ceramicnetwork/stream-tile'
import { CeramicClient } from './ceramic-http-client.js'
import { Document } from './document.js'

const a: StreamState = {
  log: [],
  type: 0,
  content: undefined,
  metadata: undefined,
  signature: SignatureStatus.GENESIS,
  anchorStatus: AnchorStatus.NOT_REQUESTED
}

const ceramic = new CeramicClient()
const d = new Document(a, new URL('http://ya.ru'), 1000000)
console.log('document created')
const b = new TileDocument(TestUtils.runningState(a), ceramic.context)
console.log('tile-document created')
```

It compiles fine. The only reason it fails _in runtime_ is because `TestUtils.runningState` calculates stream id when creating `RunningStateLike` object for optimisation purposes basically. If not for that, there is literally nothing to prevent `this.state` in TileDocument from having zero log entries.

Reply:
hmm... that's a pretty big abuse of the API though.  Note the use of `TestUtils.runningState`, that's something regular users shouldn't know about or have access to.

It seems like really the http-client should be validating that the state object it receives from the daemon is valid, but then code within TileDocument (or the other Stream handles) should be able to assume they are operating with a valid StreamState.  Validation should happen at the edges, not interspersed everywhere

Reply:
Oh, Spencer, yes, I can not agree more. I use here `TestUtils.runningState` for illustrative purposes. What is said above about it, is equally true for our `RunningState`. It is a nice coincidence that `RunningState` tries to construct `id` of a stream when constructing an object. We could have made it lazily computed though. Then, we would have no guarantees at construction time, that log is not empty.

> Validation should happen at the edges, not interspersed everywhere

Agree. Remember we had a discussion once, how (de)serialization is linked to our StateStore and shape of StreamState? I guess now it becomes clear (to me, to you, to others) how this two topics are linked.